okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/UrlSelectorImpl.java

```diff
@@ -35,9 +38,14 @@ private UrlSelectorImpl(ImmutableList<HttpUrl> baseUrls) {
         this.currentUrl = new AtomicInteger(0);
     }
 
-    static UrlSelectorImpl create(Collection<String> baseUrls) {
+    static UrlSelectorImpl create(Collection<String> baseUrls, boolean randomizeOrder) {
+        List<String> orderedUrls = new ArrayList<>(baseUrls);
+        if (randomizeOrder) {
+            Collections.shuffle(orderedUrls);
+        }
+
         ImmutableSet.Builder<HttpUrl> canonicalUrls = ImmutableSet.builder();
```

Comment:
hmm, I guess the set can make the order deterministic again.

Reply:
actually, ImmutableSet maintains the order according to javadoc

Reply:
`ImmutableSet` is order preserving, no?